push/push.go

```diff
@@ -0,0 +1,80 @@
+package push
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/provider"
+	"github.com/evcc-io/evcc/util"
+)
+
+func init() {
+	registry.AddCtx(api.Custom, NewConfigurableFromConfig)
+}
+
+// NewConfigurableFromConfig creates Messenger from config
+func NewConfigurableFromConfig(ctx context.Context, other map[string]interface{}) (Messenger, error) {
+	var cc struct {
+		Send     provider.Config
+		Encoding string
+	}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	send, err := provider.NewStringSetterFromConfig(ctx, "send", cc.Send)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewConfigurable(send, cc.Encoding)
+}
+
+// NewConfigurable creates a new Messenger
+func NewConfigurable(send func(string) error, encoding string) (*Push, error) {
+	m := &Push{
+		log:      util.NewLogger("push"),
+		send:     send,
+		encoding: strings.ToLower(encoding),
+	}
+	return m, nil
+}
+
+// Push is a configurable Messenger implementation
+type Push struct {
+	log      *util.Logger
+	send     func(string) error
+	encoding string
+}
+
+// Send implements the Messenger interface
+func (m *Push) Send(title, msg string) {
+	var res string
+
+	switch m.encoding {
+	case "json":
+		b, _ := json.Marshal(struct {
+			Title string `json:"title"`
+			Msg   string `json:"msg"`
+		}{
+			Title: title,
+			Msg:   msg,
+		})
+		res = string(b)
+	case "csv":
+		res = title + "," + msg
+	case "tsv":
+		res = title + "\t" + msg
+	case "title":
```

Comment:
this type does not "fit" into the list: json, csv, tsv are formats (so kind of an encoding), "title" and "message" (the default) are more filters. 

Wdyt to restrict encoding to "json" and "string" (or "raw") and have some second options as to whether to include the title or not ? (usually if you only use a single service endpoint and you don't need the title, you just don't define it in `events:` ...

Reply:
> this type does not "fit" into the list: json, csv, tsv are formats (so kind of an encoding), "title" and "message" (the default) are more filters.

It doesn't, but it allows returning the title instead of the message for simple command line scripts. It doesn't hurt anyway.

Reply:
As argued above, I don't buy the argument _it doesn't hurt anyway_, but your mileage may vary.